http-server-netty/src/main/java/io/micronaut/http/server/netty/multipart/MultipartBodyArgumentBinder.java

```diff
@@ -117,19 +118,21 @@ protected void doOnNext(Object message) {
                         if (LOG.isTraceEnabled()) {
                             LOG.trace("Server received streaming message for argument [{}]: {}", context.getArgument(), message);
                         }
-                        if (message instanceof ByteBufHolder && ((ByteBufHolder) message).content() instanceof EmptyByteBuf) {
+                        if (message instanceof HttpData data) {
```

Comment:
We might want to review this casttype cases too.

Reply:
the processor used here only returns HttpData anyway, so ive removed the checks.